src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Utilities/FileNameUtilties.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```

Comment:
I had to copy over all the file and path utilities from Roslyn for TargetFrameworkValueProvider. You can ignore the review for rest of the 4 files.